src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_Call.cs

```diff
@@ -512,6 +490,107 @@ private static bool IsSafeForReordering(BoundExpression expression, RefKind kind
+                // So we just return list of arguments as is.
+                for(int i = 0; i < arguments.Length; ++i)
+                {
+                    argumentKindBuilder.Add(ArgumentKind.Explicit);
+                    parameterSymbolBuilder.Add(parameters[i]);
+                }
+                return arguments;
+            }
+
+            Debug.Assert(temporariesBuilder == null);
+
+            ArrayBuilder<BoundExpression> argumentsBuilder = ArrayBuilder<BoundExpression>.GetInstance(parameters.Length);
+            ArrayBuilder<ParameterSymbol> missingParametersBuilder = ArrayBuilder<ParameterSymbol>.GetInstance(parameters.Length);
+
+            BuildExplicitArgumentsInEvaluationOrder(syntax, methodOrIndexer, expanded, argsToParamsOpt, arguments, binder.Compilation, argumentsBuilder, argumentKindBuilder, parameterSymbolBuilder, missingParametersBuilder);
+
+            AppendMissingOptionalArguments(syntax, missingParametersBuilder.ToImmutableAndFree(), binder, argumentsBuilder, argumentKindBuilder, parameterSymbolBuilder);
+
+            return argumentsBuilder.ToImmutableAndFree();
+        }
+
+        // temporariesBuilder will be null when factory is null.
+        private static bool IsRewritingRequired(
+            ref ImmutableArray<BoundExpression> rewrittenArguments,
+            Symbol methodOrIndexer,
+            bool expanded,
+            ImmutableArray<int> argsToParamsOpt,
+            bool invokedAsExtensionMethod,
+            SyntheticBoundNodeFactory factory,
+            out bool isComReceiver,
+            out ArrayBuilder<LocalSymbol> temporariesBuilder)
+        {
+            temporariesBuilder = null;
+
+            // An applicable "vararg" method could not possibly be applicable in its expanded
+            // form, and cannot possibly have named arguments or used optional parameters, 
+            // because the __arglist() argument has to be positional and in the last position. 
+
+            if (methodOrIndexer.GetIsVararg())
+            {
+                Debug.Assert(rewrittenArguments.Length == methodOrIndexer.GetParameterCount() + 1);
+                Debug.Assert(argsToParamsOpt.IsDefault);
+                Debug.Assert(!expanded);
+                isComReceiver = false;
+                temporariesBuilder = null;
+                return true;
+            }
+
+            var receiverNamedType = invokedAsExtensionMethod ?
+                                    ((MethodSymbol)methodOrIndexer).Parameters[0].Type as NamedTypeSymbol :
+                                    methodOrIndexer.ContainingType;
+
+            isComReceiver = (object)receiverNamedType != null && receiverNamedType.IsComImport;
+
+            if (factory != null)
+            {
+                temporariesBuilder = ArrayBuilder<LocalSymbol>.GetInstance();
+                rewrittenArguments = factory.MakeTempsForDiscardArguments(rewrittenArguments, temporariesBuilder);
```

Comment:
>rewrittenArguments = factory.MakeTempsForDiscardArguments(rewrittenArguments, temporariesBuilder); [](start = 16, length = 98)

I believe this is a wrong place for this operation, it should be performed before the "__arglist" check (see [https://github.com/dotnet/roslyn/issues/19382](https://github.com/dotnet/roslyn/issues/19382)). I understand that this is preexisting problem, but I suggest to move this code out of this function, into MakeArguments right before this function is called. This will make the code cleaner because we will be able to get rid of the ```factory``` and ```temporariesBuilder``` parameters, and the conditional logic associated with them. As an added bonus, the crash in the compiler will be fixed, please add the unit-test from the issue into OutVarTests.cs.